src/ApplicationRunner.php

```diff
@@ -160,39 +130,54 @@ protected function getContainer(): ContainerInterface
         return $this->container;
     }
 
+    final protected function getConfiguration(string $name): ?array
+    {
+        $config = $this->getConfig();
+        return $config->has($name) ? $config->get($name) : null;
+    }
+
     /**
      * @throws ErrorException
      */
-    protected function createDefaultConfig(): Config
+    private function createDefaultConfig(): Config
     {
-        return ConfigFactory::create(
+        $paramsGroups = [$this->paramsGroup, ...$this->nestedParamsGroups];
```

Comment:
The oreder could be reversed here. Then likely there will be no need for reverse merge modifier and internally it would look more logical.

Reply:
`Config` detect order of merge by read configuration. `$paramsGroups` need for say to `Config` that this groups need merge recursive (not reverse).